File: src/IO/dislin-11.0/java/styles.java
import de.dislin.Dislin;

public class styles {
  public static void main (String args []) {
     int i, nx, ny;

     String ctit1 = "Demonstration of CURVE";
     String ctit2 = "Line Styles";

     String ctyp [] = {"SOLID", "DOT", "DASH", "CHNDSH",
                       "CHNDOT", "DASHM", "DOTL", "DASHL"};

     float x []  = {3.f, 9.f};
     float y []  = {0.f, 0.f};

     Dislin.metafl ("cons");
     Dislin.setpag ("da4p");

     Dislin.disini ();
     Dislin.pagera ();
     Dislin.complx ();
     Dislin.center ();

     Dislin.chncrv ("both");         
     Dislin.name   ("X-axis", "x");
     Dislin.name   ("Y-axis",  "y");

     Dislin.titlin (ctit1, 1);
     Dislin.titlin (ctit2, 3);

     Dislin.graf   (0.f, 10.f, 0.f, 2.f, 0.f, 10.f, 0.f, 2.f);
     Dislin.title  ();

     for (i = 0; i < 8; i++) {
       y[0] = 8.5f - i; 
       y[1] = 8.5f - i;
       nx = Dislin.nxposn (1.0f);
       ny = Dislin.nyposn (y[0]);
       Dislin.messag (ctyp[i], nx, ny - 20);
       Dislin.curve (x, y, 2);
     }

     Dislin.disfin  ();
  }
}

File: src/IO/dislin-11.0/python/styles.py
#!/usr/bin/env python
import dislin

ctit1 = 'Demonstration of CURVE'
ctit2 = 'Line Styles'

ctyp = ['SOLID', 'DOT', 'DASH', 'CHNDSH', 
        'CHNDOT', 'DASHM', 'DOTL', 'DASHL']
x = [3., 9.]
y = [0., 0.]

dislin.metafl ('cons')
dislin.setpag ('da4p')

dislin.disini ()
dislin.pagera ()
dislin.complx ()
dislin.center ()

dislin.chncrv ('BOTH')
dislin.name   ('X-axis', 'X')
dislin.name   ('Y-axis', 'Y')

dislin.titlin (ctit1, 1)
dislin.titlin (ctit2, 3)

dislin.graf   (0., 10., 0., 2., 0., 10., 0., 2.)
dislin.title  ()

for i in range (0, 8):
  y[0] = 8.5 - i
  y[1] = 8.5 - i
  nx = dislin.nxposn (1.0)
  ny = dislin.nyposn (y[0])
  dislin.messag (ctyp[i], nx, ny - 20)
  dislin.curve  (x, y, 2)

dislin.disfin ()


